text_to_num/parsers.py

```diff
@@ -181,6 +179,297 @@ def push(self, word: str, look_ahead: Optional[str] = None) -> bool:
         return True
 
 
+class WordStreamValueParserGerman:
+    """The actual value builder engine for the German language.
+
+    The engine processes numbers blockwise and sums them up at the end.
+
+    The algorithm is based on the observation that humans gather the
+    digits by group of three to more easily speak them out.
+    And indeed, the language uses powers of 1000 to structure big numbers.
+
+    Public API:
+
+        - ``self.parse(word)``
+        - ``self.value: int``
+    """
+
+    def parse(self, word: str) -> bool:
```

Comment:
The signature says `parse` returns a `bool` but the function body has no return statement.

Reply:
The function is too long/complex.